kotlin/tests/integrationtests/com/google/idea/blaze/kotlin/KotlinSyncTest.java

```diff
@@ -0,0 +1,214 @@
+                                TargetIdeInfo.builder()
+                                        .setBuildFile(sourceRoot("src/main/kotlin/com/google/BUILD"))
+                                        .setLabel("//src/main/kotlin/com/google:lib")
+                                        .setKind("kt_jvm_library")
+                                        .addSource(sourceRoot("src/main/kotlin/com/google/Source.kotlin"))
+                                        .addSource(sourceRoot("src/main/kotlin/com/google/Other.kotlin"))
+                                        .setJavaInfo(JavaIdeInfo.builder()))
+                        .build();
+
+        setTargetMap(targetMap);
+
+        runBlazeSync(
+                new BlazeSyncParams.Builder("Sync", BlazeSyncParams.SyncMode.INCREMENTAL)
+                        .addProjectViewTargets(true)
+                        .build());
+
+        assertNoSyncErrors();
+
+        BlazeProjectData blazeProjectData =
+                BlazeProjectDataManager.getInstance(getProject()).getBlazeProjectData();
+        assertThat(blazeProjectData).isNotNull();
+        assertThat(blazeProjectData.targetMap).isEqualTo(targetMap);
+        assertThat(blazeProjectData.workspaceLanguageSettings)
+                .isEqualTo(
+                        new WorkspaceLanguageSettings(
+                                WorkspaceType.JAVA,
+                                ImmutableSet.of(LanguageClass.GENERIC, LanguageClass.KOTLIN, LanguageClass.JAVA)));
+    }
+
+    @Test
+    public void testDefaultSectionParsing() {
+        setProjectView(
+                "additional_languages:",
+                "  kotlin");
+
+        assertViewConfigState(BlazeKotlin.DEFAULT_LANGUAGE_VERSION);
+    }
+
+    @Test
+    public void testSectionParsing() {
+        setProjectView(
+                "kotlin_language_version: " + LanguageVersion.KOTLIN_1_1.getVersionString(),
+                "additional_languages:",
+                "  kotlin");
+
+        assertViewConfigState(LanguageVersion.KOTLIN_1_1);
+    }
+
+    private void assertViewConfigState(LanguageVersion languageVersion) {
+        runBlazeSync(new BlazeSyncParams.Builder("Sync", BlazeSyncParams.SyncMode.INCREMENTAL).addProjectViewTargets(true).build());
+        assertNoSyncErrors();
+
+        ProjectViewSet projectViewSet = ProjectViewManager.getInstance(getProject()).getProjectViewSet();
+        assert projectViewSet != null;
+        assertThat(BlazeKotlinSections.getLanguageLevel(projectViewSet)).isEqualTo(languageVersion);
+
+        // test the compiler reflect the project view.
+        BlazeKotlinCompilerArgumentsUpdaterCompat updater = BlazeKotlinCompilerArgumentsUpdaterCompat.build(getProject());
+        assertThat(updater.getApiVersion()).isEqualTo(languageVersion.getVersionString());
+        assertThat(updater.getLanguageVersion()).isEqualTo(languageVersion.getVersionString());
+    }
+
+
+    private void assertNoSyncErrors() {
```

Comment:
`errorCollector.assertIssues(BlazeKotlin.Issues.RULES_ABSENT_FROM_WORKSPACE)` seems to be identical to this?

Reply:
The rules being absent from the workspace is a warning `IssueOutput.warn(BlazeKotlin.Issues.RULES_ABSENT_FROM_WORKSPACE).submit(context);`. so both assertions are needed here.

Reply:
ErrorCollector doesn't make a distinction. `errorCollector.assertIssues(BlazeKotlin.Issues.RULES_ABSENT_FROM_WORKSPACE)` works just fine.

Reply:
aah I see what you mean, the underlying assertion is based on `containsExactly` `assertThat(messages).containsExactly((Object[]) requiredMessages);` I'll make the change. The method names are a misleading.